src/transform/src/predicate_pushdown.rs

```diff
@@ -656,24 +658,28 @@ impl PredicatePushdown {
-                                            func: BinaryFunc::Or,
-                                            expr1: Box::new(MirScalarExpr::CallBinary {
-                                                func: BinaryFunc::Eq,
-                                                expr1: Box::new(expr2.clone()),
-                                                expr2: Box::new(expr1.clone()),
-                                            }),
-                                            expr2: Box::new(MirScalarExpr::CallBinary {
-                                                func: BinaryFunc::And,
-                                                expr1: Box::new(MirScalarExpr::CallUnary {
-                                                    func: UnaryFunc::IsNull(func::IsNull),
-                                                    expr: Box::new(expr2.clone()),
-                                                }),
-                                                expr2: Box::new(MirScalarExpr::CallUnary {
-                                                    func: UnaryFunc::IsNull(func::IsNull),
-                                                    expr: Box::new(expr1.clone()),
-                                                }),
-                                            }),
+                                        push_downs[input].push(MirScalarExpr::CallVariadic {
+                                            func: VariadicFunc::Or,
+                                            exprs: vec![
+                                                MirScalarExpr::CallBinary {
+                                                    func: BinaryFunc::Eq,
+                                                    expr1: Box::new(expr2.clone()),
+                                                    expr2: Box::new(expr1.clone()),
+                                                },
+                                                MirScalarExpr::CallVariadic {
+                                                    func: VariadicFunc::And,
+                                                    exprs: vec![
+                                                        MirScalarExpr::CallUnary {
+                                                            func: UnaryFunc::IsNull(func::IsNull),
+                                                            expr: Box::new(expr2.clone()),
+                                                        },
+                                                        MirScalarExpr::CallUnary {
+                                                            func: UnaryFunc::IsNull(func::IsNull),
+                                                            expr: Box::new(expr1.clone()),
+                                                        },
+                                                    ],
+                                                },
+                                            ],
```

Comment:
Nit: Another opportunity to use the MirScalarExpr::eq, et. al.

Reply:
Done